core/chains/evm/logpoller/log_poller.go

```diff
@@ -707,23 +706,62 @@ func (lp *logPoller) LatestLogEventSigsAddrsWithConfs(fromBlock int64, eventSigs
 	return lp.orm.SelectLatestLogEventSigsAddrsWithConfs(fromBlock, addresses, eventSigs, confs, qopts...)
 }
 
-// GetBlocks tries to get the specified block numbers from the log pollers
-// blocks table. Returns the blocks it was able to find, empty slice if none.
-// When the log poller does not have requested blocks, it falls back
-// to RPC to fetch the missing blocks.
-// response contains blocks in the same order as "numbers" in request parameters
-// the first context parameter takes precedence over contexts passed through qopts
-func (lp *logPoller) GetBlocks(ctx context.Context, numbers []uint64, qopts ...pg.QOpt) ([]LogPollerBlock, error) {
+// GetBlocksAscendingUsingRange tries to get the specified block numbers from the log pollers
+// blocks table, and sorts them in ascending order. It falls back to the RPC for any unfulfilled
+// blocks.
+//
+// Note: it does not fetch any unrequested blocks from the RPC.
+func (lp *logPoller) GetBlocksAscendingUsingRange(ctx context.Context, numbers []uint64, qopts ...pg.QOpt) ([]LogPollerBlock, error) {
+	var blocks []LogPollerBlock
+
+	// Do nothing if no blocks are requested.
+	if len(numbers) == 0 {
+		return blocks, nil
+	}
+
+	// Sort the requested blocks in ascending order and assign them to a request map.
+	sort.Slice(numbers, func(i, j int) bool { return numbers[i] < numbers[j] })
+	blocksRequested := make(map[uint64]struct{})
+	for _, b := range numbers {
+		blocksRequested[b] = struct{}{}
+	}
+
+	// Retrieve all blocks within this range from the log poller.
 	blocksFound := make(map[uint64]LogPollerBlock)
 	qopts = append(qopts, pg.WithParentCtx(ctx))
-	lpBlocks, err := lp.orm.GetBlocks(numbers, qopts...)
+	lpBlocks, err := lp.orm.GetBlocksFromRange(numbers[0], numbers[len(numbers)-1], qopts...)
 	if err != nil {
 		lp.lggr.Warnw("Error while retrieving blocks from log pollers blocks table. Falling back to RPC...", "requestedBlocks", numbers, "err", err)
 	}
```

Comment:
Do we think it's useful if we log the block numbers we were able to get from the LP table? Something like: 
```suggestion
	else {
		lp.lggr.Debugw("Got blocks from log poller", "lpBlockNumbers", func(b []LogPollerBlock) (blockNumbers []int64) {
			for _, lpb := range b {
				blockNumbers = append(blockNumbers, lpb.BlockNumber)
			}
		}())
	}
```

We can also just pass in `lpBlocks` to the logger as a field but it might be too noisy.

Reply:
yeah for sure!